userspace/libscap/engine/bpf/scap_bpf.c

```diff
@@ -1580,6 +1633,125 @@ int32_t scap_bpf_get_stats(struct scap_engine_handle engine, OUT scap_stats* sta
+
+		for(int cpu = 0; cpu < handle->m_ncpus; cpu++)
+		{
+			struct scap_bpf_per_cpu_state v;
+			if((ret = bpf_map_lookup_elem(handle->m_bpf_map_fds[SCAP_LOCAL_STATE_MAP], &cpu, &v)))
+			{
+				*rc = scap_errprintf(handle->m_lasterr, -ret, "Error looking up local state %d", cpu);
+				return stats;
+			}
+			stats[BPF_N_EVTS].value.u64 += v.n_evts;
+			stats[BPF_N_DROPS_BUFFER_TOTAL].value.u64 += v.n_drops_buffer;
+			stats[BPF_N_DROPS_BUFFER_CLONE_FORK_ENTER].value.u64 += v.n_drops_buffer_clone_fork_enter;
+			stats[BPF_N_DROPS_BUFFER_CLONE_FORK_EXIT].value.u64 += v.n_drops_buffer_clone_fork_exit;
+			stats[BPF_N_DROPS_BUFFER_EXECVE_ENTER].value.u64 += v.n_drops_buffer_execve_enter;
+			stats[BPF_N_DROPS_BUFFER_EXECVE_EXIT].value.u64 += v.n_drops_buffer_execve_exit;
+			stats[BPF_N_DROPS_BUFFER_CONNECT_ENTER].value.u64 += v.n_drops_buffer_connect_enter;
+			stats[BPF_N_DROPS_BUFFER_CONNECT_EXIT].value.u64 += v.n_drops_buffer_connect_exit;
+			stats[BPF_N_DROPS_BUFFER_OPEN_ENTER].value.u64 += v.n_drops_buffer_open_enter;
+			stats[BPF_N_DROPS_BUFFER_OPEN_EXIT].value.u64 += v.n_drops_buffer_open_exit;
+			stats[BPF_N_DROPS_BUFFER_DIR_FILE_ENTER].value.u64 += v.n_drops_buffer_dir_file_enter;
+			stats[BPF_N_DROPS_BUFFER_DIR_FILE_EXIT].value.u64 += v.n_drops_buffer_dir_file_exit;
+			stats[BPF_N_DROPS_BUFFER_OTHER_INTEREST_ENTER].value.u64 += v.n_drops_buffer_other_interest_enter;
+			stats[BPF_N_DROPS_BUFFER_OTHER_INTEREST_EXIT].value.u64 += v.n_drops_buffer_other_interest_exit;
+			stats[BPF_N_DROPS_SCRATCH_MAP].value.u64 += v.n_drops_scratch_map;
+			stats[BPF_N_DROPS_PAGE_FAULTS].value.u64 += v.n_drops_pf;
+			stats[BPF_N_DROPS_BUG].value.u64 += v.n_drops_bug;
+			stats[BPF_N_DROPS].value.u64 += v.n_drops_buffer + \
+				v.n_drops_scratch_map + \
+				v.n_drops_pf + \
+				v.n_drops_bug;
+		}
+		i = BPF_MAX_KERNEL_COUNTERS_STATS;
+	}
+
+	/* LIBBPF STATS */
+
+	/* At the time of writing (Apr 2, 2023) libbpf stats are only available on a per program granularity.
+	 * This means we cannot measure the statistics for each filler / tail call individually.
+	 * Hopefully someone upstreams such capabilities to libbpf one day :)
+	 * Meanwhile, we can simulate perf comparisons between future LSM hooks and sys enter and exit tracepoints
+	 * via leveraging syscall selection mechanisms `handle->curr_sc_set`.
+	 */
+	if ((flags & PPM_SCAP_STATS_LIBBPF_STATS))
+	{
+		for(int bpf_prog = 0; bpf_prog < BPF_PROG_ATTACHED_MAX; bpf_prog++)
+		{
+			fd = handle->m_attached_progs[bpf_prog].fd;
+			if (fd < 0)
+			{
+				// we loop through each possible prog, landing here means prog was not attached
+				continue;
+			}
+			struct bpf_prog_info info = {};
+			__u32 len = sizeof(info);
+			if((bpf_obj_get_info_by_fd(fd, &info, &len)))
+			{
+				*rc = scap_errprintf(handle->m_lasterr, -ret, "Error getting bpf prog info for fd %d", fd);
+				continue;
+			}
+
+			for(int stat =  0;  stat < BPF_MAX_LIBBPF_STATS; stat++)
+			{
+				stats[i].type = STATS_VALUE_TYPE_U64;
```

Comment:
Do we check anywhere that `i` still fits in the buffer? How do we know that it does?